File: solutions/typescript/resistor-color/1/resistor-color.ts
// Define an array of colors corresponding to their resistance values
export const COLORS: string[] = [
  'black',   // 0
  'brown',   // 1
  'red',     // 2
  'orange',  // 3
  'yellow',  // 4
  'green',   // 5
  'blue',    // 6
  'violet',  // 7
  'grey',    // 8
  'white'    // 9
];

// Function to return the numerical value of a specified color
export const colorCode = (color: string): number => {
  const index = COLORS.indexOf(color);
  
  // If the color is not found, throw an error
  if (index === -1) {
    throw new Error('Color not found');
  }
  
  return index; // Return the index which corresponds to the numeric value
};

File: solutions/python/resistor-color/1/resistor_color.py
# Define the list of color codes
COLOR_CODES = ["black", "brown", "red", "orange", "yellow", 
               "green", "blue", "violet", "grey", "white"]

def color_code(color):
    return COLOR_CODES.index(color)

def colors():
    return COLOR_CODES

# Let's test both functions with a few examples
test_color = "green"
code_for_green = color_code(test_color)
all_colors = colors()

code_for_green, all_colors
